test/integration/targets/web3-v2/DumbContract.spec.web3.ts

```diff
@@ -0,0 +1,117 @@
+    expect(isBigNumber(withString)).to.be.true;
+    expect(withString.toString()).to.be.eq("4");
+  });
+
+  it("should allow to pass signed values in multiple ways", async () => {
+    const contract = await deployDumbContract();
+
+    const withNumber = await contract.methods.returnSigned(2).call();
+    expect(isBigNumber(withNumber)).to.be.true;
+    expect(withNumber.toString()).to.be.eq("2");
+
+    const withString = await contract.methods.returnSigned("2").call();
+    expect(isBigNumber(withString)).to.be.true;
+    expect(withString.toString()).to.be.eq("2");
+  });
+
+  it("should allow to pass address values in multiple ways", async () => {
+    const contract = await deployDumbContract();
+
+    expect(
+      await contract.methods.testAddress("0x0000000000000000000000000000000000000123").call(),
+    ).to.be.eq("0x0000000000000000000000000000000000000123");
+  });
+
+  it("should allow to pass bytes values in multiple ways", async () => {
+    const contract = await deployDumbContract();
+    const byteString = "0xabcd123456000000000000000000000000000000000000000000000000000000";
+
+    await contract.methods.callWithBytes(byteString).send({ from: accounts[0] });
+
+    const result = await contract.methods.byteArray().call();
+    expect(result).to.be.a("string");
+    expect(result).to.eq(byteString);
+  });
+
+  it("should allow to pass Buffer for dynamic bytes array", async () => {
+    const contract = await deployDumbContract();
+    const byteString = "0xabcd123456000000000000000000000000000000000000000000000000000000";
+
+    await contract.methods.callWithDynamicByteArray(byteString).send({ from: accounts[0] });
+
+    const result = await contract.methods.dynamicByteArray().call();
+    expect(result).to.be.a("string");
+    expect(result).to.eq(byteString);
+  });
+
+  it("should allow to pass boolean values", async () => {
+    const contract = await deployDumbContract();
+
+    const res = await contract.methods.callWithBoolean(true).call();
+    expect(res).to.be.deep.eq(true);
+  });
+
+  it("should allow to pass numeric arrays values in multiple ways", async () => {
+    const contract = await deployDumbContract();
+
+    const res = await contract.methods.callWithArray2(["1", 2]).call();
+    expect(res.length).to.be.eq(2);
+    expect(isBigNumber(res[0])).to.be.true;
+    expect(isBigNumber(res[1])).to.be.true;
+    expect(res[0].toString()).to.be.eq("1");
+    expect(res[1].toString()).to.be.eq("2");
+  });
+
+  it("should allow to pass strings ", async () => {
+    const contract = await deployDumbContract();
+
+    expect(await contract.methods.testString("abc").call()).to.be.deep.eq("abc");
+  });
+
+  it("should allow to clone contracts ", async () => {
+    const contract = await deployDumbContract();
+
+    const contractClone = await contract.clone();
+
+    expect(contractClone).not.to.be.eq(contract);
+    // web3@2.0.0 just shallow copies the contract and `address` is `undefined`
```

Comment:
@krzkaczor @quezak - Just wanted to point this one. The cloned contract doesn't have same `address` as that of the original contract. Hence validating the `options.data` equality.